### DISCUSSION TITLE
Is it a bug of plugin?

### DISCUSSION THREAD

**Discussion by UniquePerch** [antlr/intellij-plugin-v4#645] [Q&A]
For example:
I have a parser defination of [seq]:

![image](https://github.com/antlr/intellij-plugin-v4/assets/92765498/1962dd5c-5bfc-486d-9e22-2e6d64ee09ee)
Notice the last defination of [seq] : seq_function SEMI?

It has been defined as followd:

![image](https://github.com/antlr/intellij-plugin-v4/assets/92765498/980a88cf-8259-4dac-9215-34f62c2cef16)

I also have a lexer defination of [POINT_D] (defined in another g4 file):
![image](https://github.com/antlr/intellij-plugin-v4/assets/92765498/578b3e70-9de6-47dc-b1e3-7c23595e1be8)

Then I import a whole lexer defination（However,if I use tokenVocab to associated lexer file, it also happens）,use the plugin to generate parse tree(input:"seq(1,2)"), something  unexpected happened , plugin regard my input "(1,2)" as a POINT_D lexer.
![image](https://github.com/antlr/intellij-plugin-v4/assets/92765498/c0ed6460-69ec-4c52-a7ae-a36ab3f89b48)

I don't know whether it is my mistake or it is just a bug of this plugin?

**Comment by bjansen** [comment]
Hi,

In your current grammar, the input `(1,2)` can match either one of these:
* `POINT_D`
* `LR_BRACKET seq_expression RR_Bracket`

But because the lexing phase happens before the parsing phase, ANTLR will match it to a `POINT_D`. Then it will complain that it was expecting a `seq_expression` but got a `POINT_D` instead.

I think you should rework your grammar to transform your `POINT_D` _lexing_ rule to a _parsing_ rule.
